sources/main.cpp

```diff
@@ -209,19 +209,32 @@ int main(int argc, char* argv[]) {
 
     // reset combination was pressed on startup
     if (buttonHandler->resetButtonsPressed()) {
-        if (QFile::exists(qEnvironmentVariable(Environment::ENV_YIO_APP_DIR, appPath) + "/config.json") &&
-            QFile::exists("/boot/config.json")) {
+        if (QFile::exists(qEnvironmentVariable(Environment::ENV_YIO_APP_DIR, appPath) + "/config.json")) {
             // create marker file
             QFile file("/firstrun");
             if (file.open(QIODevice::WriteOnly)) {
                 file.close();
 
+                // remove old config if exists
+                if (QFile::exists("/boot/config.json.old")) {
+                    if (!QFile::remove("/boot/config.json.old")) {
+                        qCCritical(CLASS_LC) << "Error removing old configuration file.";
+                    }
+                }
+
                 // rename existing config
-                QFile::rename("/boot/config.json", "/boot/config.json.old");
+                if (!QFile::rename("/boot/config.json", "/boot/config.json.old")) {
+                    qCCritical(CLASS_LC) << "Error backing up exiting configuration.";
+                }
```

Comment:
typo: `existing`

There's no reason to continue, since the QFile::copy further down will fail because the destination file still exists...

The bigger issue is the classical race condition though: we remove an essential file first and then copy another file at it's place. If anything goes wrong in between we have an inconsistent state!
The safe way would be:
1. Backup /boot/config.json without removing it
1. Overwrite /boot/config.json with the default config

Unfortunately Qt doesn't provide an overwrite mode for QFile::copy 👎 
So we have to use good old C or modern C++, e.g. https://en.cppreference.com/w/cpp/filesystem/copy_file
which has a `copy_options::overwrite_existing` flag 👍